core/templates/dev/head/exploration_editor/main_tab/StateResponses.js

```diff
@@ -306,6 +308,26 @@ oppia.controller('StateResponses', [
         return choiceIndices.every(function(choiceIndex) {
           return handledAnswersArray.indexOf(choiceIndex) !== -1;
         });
+      } else if (interactionId === 'ItemSelectionInput') {
+        var numChoices = $scope.getAnswerChoices().length;
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
```

Comment:
Having serious problems trying to understand this. So there would be an array 
```
//array of booleans for equals, containsOne, omitsOne
        var choiceRules = [false, false, false];
```
and then in the For loops, if ruleSpecs[j].rule_type is ContainsAtLeastOneOf and ruleSpecs[j].inputs.x.0 is ????

I don't understand what's supposed to happen then.